sycl/test-e2e/AddressSanitizer/out-of-bounds/common/kernel-demangle.cpp

```diff
@@ -0,0 +1,20 @@
+// REQUIRES: linux, cpu, aspect-fp64
+// RUN: %{build} %device_sanitizer_flags -O2 -g -o %t
+// RUN: env SYCL_PREFER_UR=1 %{run} not %t &> %t.txt ; FileCheck --input-file %t.txt %s
+#include <sycl/sycl.hpp>
+
+int main() {
+  sycl::queue Q;
+  constexpr std::size_t N = 123456;
+  auto *array = sycl::malloc_device<double>(N, Q);
+
+  Q.submit([&](sycl::handler &h) {
+    h.parallel_for<class MyKernel>(
+        sycl::nd_range<1>(N + 1, 1),
+        [=](sycl::nd_item<1> item) { ++array[item.get_global_id(0)]; });
+  });
+  Q.wait();
+  // CHECK: kernel <typeinfo name for main::{lambda(sycl::_V1::handler&)#1}::operator()(sycl::_V1::handler&) const::MyKernel>
```

Comment:
What I understand is that this test is testing out of bounds accesses. The error message when that happens is this? I would expect something like `Out of bounds access in <kernel_name>`.

Reply:
Detecting out-of-bounds errors are already tested in other folders, this test is used for testing if kernel name is demangled in "kernel <kernel_name>" only.

Reply:
And how's that related to the topic of the PR (i.e. sycl::free related errors)?

Reply:
Well, it's not related to this PR. I add this test for https://github.com/oneapi-src/unified-runtime/pull/1355.

Reply:
Right, then I would advice to make a separate PR for this test.

Reply:
okay

Reply:
I submit PR https://github.com/intel/llvm/pull/12948 for this test now.